web-services/query/src/main/java/datawave/webservice/query/logic/composite/CompositeQueryLogic.java

```diff
@@ -137,31 +160,45 @@ public CompositeQueryLogic() {}
-        this.queryLogics = new ArrayList<>(other.queryLogics);
+        if (other.queryLogics != null) {
+            this.queryLogics = new ArrayList<>();
+            for (QueryLogic<?> logic : other.queryLogics) {
+                try {
+                    logic = (QueryLogic) logic.clone();
+                } catch (CloneNotSupportedException e) {
+                    throw new RuntimeException(e);
+                }
+                this.queryLogics.add(logic);
+            }
+        }
+        this.allMustInitialize = other.allMustInitialize;
+        this.checkTables = other.checkTables;
     }
     
     @Override
     public GenericQueryConfiguration initialize(Connector connection, Query settings, Set<Authorizations> runtimeQueryAuthorizations) throws Exception {
         
-        for (QueryLogic<?> logic : queryLogics) {
-            final QueryLogic<?> queryLogic = logic;
-            int count = CollectionUtils.countMatches(
-                            queryLogics,
-                            object -> {
-                                if (object instanceof QueryLogic<?>) {
-                                    if (queryLogic.getClass().equals(((QueryLogic<?>) object).getClass())
-                                                    && queryLogic.getTableName().equals(((QueryLogic<?>) object).getTableName())) {
-                                        return true;
+        if (checkTables) {
+            for (QueryLogic<?> logic : queryLogics) {
+                final QueryLogic<?> queryLogic = logic;
+                int count = CollectionUtils.countMatches(
+                                queryLogics,
+                                object -> {
+                                    if (object instanceof QueryLogic<?>) {
+                                        if (queryLogic.getClass().equals(((QueryLogic<?>) object).getClass())
+                                                        && queryLogic.getTableName().equals(((QueryLogic<?>) object).getTableName())) {
+                                            return true;
+                                        } else {
+                                            return false;
+                                        }
                                     } else {
                                         return false;
                                     }
-                                } else {
-                                    return false;
-                                }
-                            });
-            
-            if (count > 1) {
-                throw new RuntimeException("More than one instance of query logic class configured with the same table: " + logic.getClass().getName());
+                                });
+                
+                if (count > 1) {
+                    throw new RuntimeException("More than one instance of query logic class configured with the same table: " + logic.getClass().getName());
+                }
```

Comment:
This is an unusual way to perform this check. Not only that, `CollectionUtils.countMatches` is deprecated. 

How about we loop over the list of query logics putting a composite of their class/table name into a set. If the set's `add` method returns false, we can throw our `RuntimeException`. That way we don't have to traverse the `queryLogicsTable `n-squared times (once in the loop and then once for each item in the loop in `CollectionUtils.countMatches()`)

Would something like this achieve the same goal?
```java
        if (checkTables) {
            Set<String> dupeCheckSet = new HashSet<>();
            for (QueryLogic<?> logic : queryLogics) {
                final String key = logic.getClass().getName() + ":" + logic.getTableName();
                if (!dupeCheckSet.add(key)) {
                    throw new RuntimeException("More than one instance of query logic class configured with the same table: " + key);
                }
            }
        }
```

Reply:
I am removing this whole check tables thing entirely.  I realized that was only there to try and create uniqueness in the logicStates tree map (see the comparator that was at the top of the class).  Very hokey concept to start with.